catx/tree.py

```diff
@@ -104,15 +104,19 @@ def __init__(
         self.tree_params = tree_params
 
         self.networks = {
-            depth: network_builder.create_network(depth=depth)
+            depth: custom_network(depth=depth)
             for depth in range(self.tree_params.depth)
         }
 
-    def __call__(self, obs: Array) -> Dict[int, Logits]:
+    def __call__(
+        self, obs: Array, key: PRNGKey, network_extras: NetworkExtras
```

Comment:
The `key` is passed directly to the networks, but the networks, being `haiku.Modules` can get the key by using `hk.next_rng_key()`. Could we remove the `key` parameter?